schema/dex/trades.sql

```diff
@@ -157,6 +158,51 @@ WITH rows AS (
 
         UNION
 
+        --- Kyber_V2
+        SELECT
+            evt_block_time AS block_time,
+            'Kyber' AS project,
+            NULL AS VERSION,
+            src AS token_a_address,
+            dest AS token_b_address,
+            kyber_v2."Network_evt_KyberTrade".contract_address exchange_contract_address,
+            evt_tx_hash AS tx_hash,
+            ethereum.transactions.from AS trader_a,
+            NULL::bytea AS trader_b,
+            NULL::integer[] AS trace_address,
+            evt_index AS evt_index,
+
+            (SELECT SUM(a)
+                    FROM UNNEST(CASE
+                                    WHEN src_token.contract_address = '\xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee' THEN "e2tSrcAmounts"
+                                    ELSE "t2eSrcAmounts"
+                                END) AS a) AS token_a_amount_raw,
+
+            (SELECT SUM(s)
+            FROM UNNEST(ARRAY
+                            (SELECT CASE
+                                        WHEN dst_token.contract_address = '\xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee' THEN -- trade from token - eth, then dst_amount should be eth
+                                                (SELECT CASE
+                                                            WHEN dst_token.decimals >= src_token.decimals
+                                                            THEN a*b*power(10, (dst_token.decimals-src_token.decimals))/1e18
+                                                            ELSE (a*b) / (1e18 * power(10, (src_token.decimals - dst_token.decimals)))
+                                                        END
+                                                FROM unnest("t2eSrcAmounts", "t2eRates") AS t(a,b))
+                                        ELSE -- trade from eth - token or token - token, dst_amount should be token
+                                                (SELECT CASE
+                                                            WHEN dst_token.decimals >= src_token.decimals
+                                                            THEN a*b*power(10, (dst_token.decimals-src_token.decimals))/1e18
+                                                            ELSE (a*b) / (1e18 * power(10, (src_token.decimals - dst_token.decimals)))
+                                                        END
+                                                FROM unnest("e2tSrcAmounts", "e2tRates") AS t(a,b))
+                                    END)) s) AS token_b_amount_raw
+        FROM kyber_v2."Network_evt_KyberTrade"
+        JOIN erc20."tokens" src_token ON kyber_v2."Network_evt_KyberTrade".src = src_token.contract_address
+        JOIN erc20."tokens" dst_token ON kyber_v2."Network_evt_KyberTrade".dest = dst_token.contract_address
+        JOIN ethereum.transactions ON ethereum.transactions.hash = kyber_v2."Network_evt_KyberTrade".evt_tx_hash;
```

Comment:
Remove `;` in the end here.

Joining with Ethereum transactions hurts performance but it might be needed...?

Reply:
maybe @mewwts has an opinion on whether performance is an issue here?

Reply:
Yeah, I am joining ethereum.transactions here to get the function caller, I assume it is the trader. Is there any better way to get that address?

Reply:
`ethereum.transactions."from"` will get you the external account that initiated the transaction
`ethereum.traces."from"` will get you the account/contract that called the function

Reply:
@mewwts the account that initiated the transaction is enough for us, using traces seems not improve performance but also bring more complicated we do not want.